compiler/rustc_target/src/spec/armv8r_none_eabihf.rs

```diff
@@ -0,0 +1,32 @@
+// Targets the Little-endian Cortex-R52 processor (ARMv8-R)
+
+use crate::spec::{Cc, LinkerFlavor, Lld, PanicStrategy, RelocModel, Target, TargetOptions};
+
+pub fn target() -> Target {
+    Target {
+        llvm_target: "armv8r-none-eabihf".into(),
+        pointer_width: 32,
+        data_layout: "e-m:e-p:32:32-Fi8-i64:64-v128:64:128-a:0:32-n32-S64".into(),
+        arch: "arm".into(),
+
+        options: TargetOptions {
+            abi: "eabihf".into(),
+            linker_flavor: LinkerFlavor::Gnu(Cc::No, Lld::Yes),
+            linker: Some("rust-lld".into()),
+            relocation_model: RelocModel::Static,
+            panic_strategy: PanicStrategy::Abort,
+            // The Cortex-R52 supports either fp-armv8 single-precision only with 16
+            // double-precision registers, or neon-fp-armv8 with 32 double-precision registers.
+            // Use the lesser of these two options as the default.
+            //
+            // Reference:
+            // Arm Cortex-R52 Processor Technical Reference Manual - Chapter 15 Advanced SIMD and floating-point support
+            features: "+fp-armv8,-d32,-fp64".into(),
+            max_atomic_width: Some(64),
+            emit_debug_gdb_scripts: false,
+            // GCC and Clang default to 8 for arm-none here
```

Comment:
I was confused about this, so I investigated. You get C compilers produce 1-byte large enums only with the `-fshort-enums` flag on armv8r targets, rather than by default, so this comment is not quite correct. It seems like it might have been added (to armv7r, which follows this behaviour) by mistake or propagated from some other targets.

Reply:
Hrm... I just experimented and stock `gcc` will give the size of an enum as one byte without using `-fshort-enums`:
```
$ arm-none-eabi-gcc --version
arm-none-eabi-gcc (Arm GNU Toolchain 12.2 (Build arm-12-mpacbti.34)) 12.2.1 20230214
Copyright (C) 2022 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
$ cat foo.c
enum foo {
    FOO,
};

unsigned long f(void) {
    return sizeof(enum foo);
}
$ arm-none-eabi-gcc -march=armv8-r -c -o foo.o -Os foo.c
$ arm-none-eabi-objdump -d foo.o

foo.o:     file format elf32-littlearm


Disassembly of section .text:

00000000 <f>:
   0:	e3a00001 	mov	r0, #1
   4:	e12fff1e 	bx	lr
```

However, the same experiment with `clang` 16.0.6 and `-target armv8r-none-eabi` shows `4`...

Reply:
Looks like AAPCS explicitly allows either convention: https://github.com/ARM-software/abi-aa/blob/main/aapcs32/aapcs32.rst#enumerated-types

Reply:
Looks like all other arm*-none-eabi* targets in Rust are using the small enum ABI, though the discrepancy with recent `clang` is a little worrying. Will have to do some archaeology to understand if/when `clang` ever had this default, as the comments claim.

Reply:
Yea, I only tested clang myself, so I sort-of assumed it might be similar with GCC. Either way I imagine these targets are almost exclusively used together with a gcc of some shape or form, so leaving it as `Some(8)` is fine in that case.

(Although amending a comment to that effect would be nice)

Reply:
Happily. `gcc` has done this for arm-none-eabi going back to 2004-ish, and changed the  the `arm-linux` targets use `int` enum starting in 2005. I was unable to find evidence that `clang` ever had default `short-enum` for any Arm targets, though it does use it for some other targets.

Reply:
Updated all the comments claiming this that I could find, and removed one instance of this option and its comment that was not necessary due to using `thumb_base`.